### ISSUE TITLE
Any way to match a wrapper function?

### ISSUE THREAD

**Issue by Martaver** [Microsoft/typescript-styled-plugin#40]
One nasty parts of using `styled-components` with TS is the lack of typing when you pass props to your style literal.

One solution can be found in `styled-components-ts` which wraps the `styled-components` function to expose correct types for props. Where you get:
```
const StyledComponents = styledComponentWithProps<MyProps>(styled.div)`
    ${props => props.myValue}
`
```

And there's no need to write `${(props: MyProps) => {...}}` with type annotations every time.

Only problem is that this fairly neatly breaks vscode's syntax highlighting and typescript-styled-plugin.

Any way to resolve this...? I've tried adding `styledComponentWithProps`  to tags under the extension configuration in tsconfig... no joy...

**Comment by Martaver** [comment]
I guess at the core, this is the issue: https://github.com/Microsoft/TypeScript/issues/11947

However, I was wondering if there was any magic hack to make `typescript-styled-plugin` pick up on the string anyway?

**Comment by mjbvz** [comment]
Does #56 solve this issue?

**Comment by mjbvz** [comment]
Closing as potentially addressed / configurable with #56
